src/other/tables.jl

```diff
@@ -11,27 +11,32 @@ Tables.materializer(df::AbstractDataFrame) = DataFrame
 
 getvector(x::AbstractVector) = x
 getvector(x) = collect(x)
-fromcolumns(x) = DataFrame(AbstractVector[getvector(c) for c in Tables.eachcolumn(x)], Index(collect(Symbol, propertynames(x))))
+fromcolumns(x; copycolumns::Bool=true) =
+    DataFrame(AbstractVector[getvector(c) for c in Tables.eachcolumn(x)],
+              Index(collect(Symbol, propertynames(x))),
+              copycolumns=copycolumns)
 
-function DataFrame(x)
+function DataFrame(x; copycolumns::Bool=true)
     if x isa AbstractVector && all(col -> isa(col, AbstractVector), x)
-        return DataFrame(Vector{AbstractVector}(x))
+        return DataFrame(Vector{AbstractVector}(x), copycolumns=copycolumns)
     end
     if applicable(iterate, x)
         if all(v -> v isa Pair{Symbol, <:AbstractVector}, x)
-            return DataFrame(AbstractVector[last(v) for v in x], [first(v) for v in x])
+            return DataFrame(AbstractVector[last(v) for v in x], [first(v) for v in x],
+                             copycolumns=copycolumns)
         end
     end
     if Tables.istable(x)
-        return fromcolumns(Tables.columns(x))
+        return fromcolumns(Tables.columns(x), copycolumns=copycolumns)
     end
     throw(ArgumentError("unable to construct DataFrame from $(typeof(x))"))
 end
 
-Base.append!(df::DataFrame, x) = append!(df, DataFrame(x))
+Base.append!(df::DataFrame, x) = append!(df, DataFrame(x, copycolumns=false))
```

Comment:
I am not sure what `x` was intended to be?

Reply:
Any table I guess. Sounds safe anyway. Could also use `convert`.

Reply:
@quinnj?

Reply:
I would say no keyword argument here, since we'll be appending anyway, so copy would just be wasteful.